src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/OrderingHelper.cs

```diff
@@ -0,0 +1,281 @@
+            }
+
+            var orderedChildren = parent.Children.Where(x => IsValidDisplayOrder(GetDisplayOrder(x))).OrderBy(x => GetDisplayOrder(x)).ToArray();
+
+            for (var i = 0; i < orderedChildren.Length; ++i)
+            {
+                var sibling = orderedChildren[i];
+                if (GetDisplayOrder(sibling) == displayOrder)
+                {
+                    return returnSibling(i, orderedChildren);
+                }
+            }
+
+            return null;
+        }
+
+        private static IProjectTree2 PreviousSibling(IProjectTree projectTree)
+        {
+            return GetSiblingByDisplayOrder(projectTree, (i, orderedChildren) =>
+            {
+                if (i == 0)
+                {
+                    return null;
+                }
+
+                return orderedChildren.ElementAt(i - 1) as IProjectTree2;
+            });
+        }
+
+        private static IProjectTree2 NextSibling(IProjectTree projectTree)
+        {
+            return GetSiblingByDisplayOrder(projectTree, (i, orderedChildren) =>
+            {
+                if (i == (orderedChildren.Count() - 1))
+                {
+                    return null;
+                }
+
+                return orderedChildren.ElementAt(i + 1) as IProjectTree2;
+            });
+        }
+
+        private static IProjectTree GetSiblingByMoveAction(IProjectTree projectTree, MoveAction moveAction)
+        {
+            switch (moveAction)
+            {
+                case MoveAction.Up:
+                    return PreviousSibling(projectTree);
+
+                case MoveAction.Down:
+                    return NextSibling(projectTree);
+            }
+
+            return null;
+        }
+
+        private static ProjectItemElement GetReferenceElementByMoveAction(UnconfiguredProject unconfiguredProject, Project project, IProjectTree projectTree, MoveAction moveAction)
+        {
+            switch (moveAction)
+            {
+                case MoveAction.Up:
+                    return GetElements(unconfiguredProject, project, projectTree).FirstOrDefault();
+
+                case MoveAction.Down:
+                    return GetElements(unconfiguredProject, project, projectTree).LastOrDefault();
+            }
+
+            return null;
+        }
+
+        private static void InsertChildByMoveAction(ProjectElementContainer parent, ProjectItemElement element, ProjectItemElement referenceElement, MoveAction moveAction)
+        {
+            switch (moveAction)
+            {
+                case MoveAction.Up:
+                    parent.InsertBeforeChild(element, referenceElement);
+                    break;
+
+                case MoveAction.Down:
+                    parent.InsertAfterChild(element, referenceElement);
+                    break;
+            }
+        }
+
+        private static async Task<bool> Move(ConfiguredProject configuredProject, IProjectTree projectTree, MoveAction moveAction)
```

Comment:
Consider naming this `MoveAsync`

Reply:
That's fair. In that case I'll rename the others to Async as well.